tests/string_test.c

```diff
@@ -81,3 +81,23 @@ static int binary_string_test_fn(struct aws_allocator *alloc, void *ctx) {
     aws_string_destroy((void *)binary_string);
     return 0;
 }
+
+AWS_TEST_CASE(string_compare_test, string_compare_test_fn);
+static int string_compare_test_fn(struct aws_allocator *alloc, void *ctx) {
+    AWS_STATIC_STRING_FROM_LITERAL(str_1, "foo");
```

Comment:
name these strings after their contents to make it easier to scan the test cases